File: password-handler-back-end/src/crypto/aes.js
const crypto = require('crypto');

class AesEncryption {
    static #BLOCK_SIZE = 256;
    static #AES_ALGORITHM = 'aes-'+AesEncryption.#BLOCK_SIZE.toString()+'-cbc';

    static decryptData(encryptedData, key, iv) {
        const decipher = crypto.createDecipheriv(AesEncryption.#AES_ALGORITHM, key, iv);
        let decryptedData = Buffer.concat([decipher.update(encryptedData, 'hex'), decipher.final()]);
        return decryptedData;
    }

    static encryptData(decryptedData, key, iv) {
        const cipher = crypto.createCipheriv(AesEncryption.#AES_ALGORITHM, key, iv);
        let encryptedData = Buffer.concat([cipher.update(decryptedData), cipher.final()]);
        return encryptedData;
    }

    static generateKey() {
        return crypto.randomBytes(AesEncryption.#BLOCK_SIZE/8);
    }

    static generateIv() {
        return crypto.randomBytes(16);
    }
}

module.exports = AesEncryption;

File: password-handler-back-end/src/crypto/aes.py
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes

def encrypt_data(decryptedData, key, iv):
    cipher = AES.new(key, AES.MODE_EAX, iv)
    encrypted_data = cipher.encrypt(decryptedData)
    return encrypted_data

def decrypt_data(encryptedData, key, iv):
    decipher = AES.new(key, AES.MODE_EAX, iv)
    decrypted_data = decipher.decrypt(encryptedData)
    return decrypted_data

def generate_key():
    return get_random_bytes(32)

def generate_iv():
    return get_random_bytes(16)


